### PR TITLE
installer.sh.in: fixes and improvements

### PR DESCRIPTION
**Pull request by pullmoll**
+ Add INFOBOX() because dialog --infobox is incompatible with --keep-tite on xterm
+ Use clear after INFOBOX() to not leave irritating info on the screen
+ Make the live system copy process use a --progressbox
+ Use a tar pipe to copy from live ISO to target file system

### CODE REVIEW

**Comment by emdete** [comment]
two remarks about your PR: 

- it's probably better to have a PR per enhancement.

- the `tput clear` is different from the aproach used by the script already (`clear` in two places), one should stick to one solution for one problem.

File: installer.sh.in [hunk 1]
```diff
@@ -93,6 +93,13 @@ DIALOG() {
     return $?
 }
 
+INFOBOX() {
+    # Note: dialog --infobox and --keep-tite don't work together
```
Comment (Vaelatern) [inline installer.sh.in]: Is it `--keep-title`?

File: installer.sh.in [hunk 2]
```diff
@@ -596,10 +603,10 @@ menu_rootpassword() {
     local _firstpass _secondpass _desc
 
     while true; do
-        if [ -n "${_firstpass}" ]; then
-            _desc="Enter the root password again"
-        else
+        if [ -z "${_firstpass}" ]; then
             _desc="Enter the root password"
+        else
+            _desc="$_desc again"
```
Comment (Vaelatern) [inline installer.sh.in]: :+1:

File: installer.sh.in [hunk 3]
```diff
@@ -610,9 +617,9 @@ menu_rootpassword() {
             fi
             if [ -n "${_firstpass}" -a -n "${_secondpass}" ]; then
                 if [ "${_firstpass}" != "${_secondpass}" ]; then
-                    DIALOG --infobox "Passwords do not match! Please enter again." 6 80
+                    INFOBOX "Passwords do not match! Please enter again." 6 60
```
Comment (Vaelatern) [inline installer.sh.in]: Why the change from 80 columns wide?

File: installer.sh.in [hunk 4]
```diff
@@ -644,8 +654,10 @@ menu_useraccount() {
     done
 
     while true; do
+        _preset=$(get_option USERNAME)
+        [ -z "$_preset" ] && _preset="Void User"
         DIALOG --inputbox "Enter a user name for login '$(get_option USERLOGIN)' :" \
-            ${INPUTSIZE} "$(get_option USERLOGIN)"
+            ${INPUTSIZE} "$_preset"
```
Comment (Vaelatern) [inline installer.sh.in]: :+1: to this and the use of `$_preset` above

File: installer.sh.in [hunk 5]
```diff
@@ -1087,13 +1100,32 @@ umount_filesystems() {
...
-    LANG=C cp -axvnu / $TARGETDIR >$LOG 2>&1
+    TITLE="Check $LOG for details ..."
+    INFOBOX "Counting files, please be patient ..." 4 60
+    copy_total=$(tar -c -v --one-file-system -f /dev/null / 2>/dev/null | wc -l)
+    export copy_total copy_count=0 copy_percent=
+    tar -c --one-file-system -f - / 2>/dev/null | \
+        tar -x -v -p -f - -C $TARGETDIR | \
+        log_and_count | \
+        DIALOG --title "${TITLE}" \
+            --progressbox "Copying live image to target rootfs." 5 60
```
Comment (Vaelatern) [inline installer.sh.in]: Uncertain about the second tar's use of `--one-file-system`

File: installer.sh.in (hunk 3 above)
Comment (Vaelatern) [inline installer.sh.in]: Nevermind. Noticed that this is consistent style with the rest of the installer.

File: installer.sh.in (hunk 5 above)
Comment (Vaelatern) [inline installer.sh.in]: Ah, nvm, understand how that's working now

Comment (Vaelatern) [inline installer.sh.in]: For consistency, might those flags share a variable, for `copy_total` and the actual copy?

File: installer.sh.in (hunk 1 above)
Comment (pullmoll) [inline installer.sh.in]: It is --keep-tite (sic!). See man dialog(1).

File: installer.sh.in (hunk 5 above)
Comment (pullmoll) [inline installer.sh.in]: I can store them in e.g. tar_flags, yes.

**Comment by pullmoll** [comment]
@emdete In this case the `INFOBOX` enhancement came as a side effect of using the `DIALOG` function. Both changes are for improving the user experience when running void-setup and specificially running it in an xterm.

I changed the `tput clear` to just `clear`, thanks!

File: installer.sh.in (hunk 1 above)
Comment (Vaelatern) [inline installer.sh.in]: Go figure!

File: installer.sh.in (hunk 5 above)
Comment (Vaelatern) [inline installer.sh.in]: Yes, a necessary difference, but the rest can be put in a var when the same across invocations.

**Comment by pullmoll** [comment]
Now uses `clear` instead of `tput clear` everywhere. Progress is now updated in per mill steps.

File: installer.sh.in [hunk 6]
```diff
@@ -93,6 +93,13 @@ DIALOG() {
     return $?
 }
 
+INFOBOX() {
+    # Note: dialog --infobox and --keep-tite don't work together
+    dialog --colors --no-shadow --no-mouse \
+        --backtitle "${BOLD}${WHITE}Void Linux installation -- http://www.voidlinux.eu/ (@@MKLIVE_VERSION@@)${RESET}" \
+        --title "${TITLE}" --aspect 20 --infobox "$@"
```
Comment (Vaelatern) [inline installer.sh.in]: With the quotes around `$@`, are the width and height actually expanded to be two args as they should?

Comment (pullmoll) [inline installer.sh.in]: The `"$@"` expands to a list of quoted args, not one list of args enclosed in quotes. Is this is bash or POSIX sh, I don't know. However, if you look at `DIALOG()` it does the same.

**Comment by pullmoll** [comment]
There seem no more objections and the change is known to fix #82 thus I'll merge now.
